stub-docs.rb

```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
+
+debug = false
+
+new_folder = 'documented'
+FileUtils.rm_rf(new_folder) if File.exist?(new_folder)
+
+files = []
+# files = ['HPXMLtoOpenStudio/resources/hpxml.rb']
+['BuildResidentialHPXML', 'BuildResidentialScheduleFile', 'HPXMLtoOpenStudio', 'ReportSimulationOutput', 'ReportUtilityBills'].each do |folder|
+  files += Dir[File.join(folder, 'resources/*.rb')]
+end
+
+def get_params(lines, idx)
+  params = []
+  descs = []
+  ret = nil
+  lines[0..idx - 1].reverse.each_with_index do |line, i|
+    if line.include?('# @param')
+      l = line.strip.split(' ')
+      params << [l[2], l[3], l[4..-1].join(' ')]
+    elsif line.include?('# @return')
+      l = line.strip.split(' ')
+      ret = [l[2], l[3..-1].join(' ')]
+    elsif line.strip == '#' # blank line
+      # no-op
+    elsif line.strip.start_with?('#')
+      descs << line.strip
+    else
+      return params, descs, ret, idx - i
+    end
+  end
+end
+
+def set_params(lines, idx, params, descs, ret, tab, is_method)
+  if is_method
+    if ret.nil?
+      lines.insert(idx, "#{tab}# @return [TODO] TODO")
+    else
+      lines.insert(idx, "#{tab}# @return #{ret[0]} #{ret[1]}")
+    end
+    params.each do |param|
+      lines.insert(idx, "#{tab}# @param #{param[0]} #{param[1]} #{param[2]}")
+    end
+    lines.insert(idx, "#{tab}#")
+  end
+  descs = ['# TODO'] if descs.empty?
+  descs.each do |desc|
+    lines.insert(idx, "#{tab}#{desc}")
+  end
+  return lines
+end
+
+def common_parameters(param, type, desc)
+  params = [['model', '[OpenStudio::Model::Model]', 'model object'],
+            ['runner', '[OpenStudio::Measure::OSRunner]', 'runner object'],
+            ['hpxml', '[HPXML]', 'hpxml object'],
+            ['hpxml_bldg', '[HPXML::Building]', 'individual HPXML Building dwelling unit object'],
+            ['weather', '[WeatherProcess]', 'TODO'],
+            ['epw_file', '[OpenStudio::EpwFile]', 'TODO'],
+            ['space', '[OpenStudio::Model::Space]', 'OpenStudio Space object'],
+            ['surface', '[OpenStudio::Model::Surface]', 'OpenStudio Surface object']]
+  params.each do |p|
+    next if p[0] != param
+
+    type = p[1] if type == '[TODO]'
+    desc = p[2] if type == 'TODO'
+  end
+  return type, desc
+end
+
+files.each do |file|
+  next if file.include?('hpxml.rb')
+
+  puts "File: #{file}"
+
+  lines = File.readlines(file)
+
+  new_path = File.join(new_folder, file)
+  new_parent_folder = File.dirname(new_path)
+  FileUtils.mkdir_p(new_parent_folder) if !File.exist?(new_parent_folder)
+
+  # previous_classes = ObjectSpace.each_object(Class).to_a
+  previous_classes = []
+
+  require_relative file
+
+  new_classes = ObjectSpace.each_object(Class).to_a - previous_classes
```

Comment:
Admittedly, running into trouble using this approach of trying to require the file and then get classes/methods out of it. Repeated/nested class/method names cause issues. Instead of getting the script to operate perfectly, it may be a more effective use of time to take over with manual edits.

Reply:
Go with whatever it easiest. I assume the script is temporary and will be deleted before merging.

Reply:
Yes, I was only planning to use it as a temporary helper script.